cranelift/interpreter/src/frame.rs

```diff
@@ -2,41 +2,45 @@
 
 use cranelift_codegen::data_value::DataValue;
 use cranelift_codegen::ir::{Function, Value as ValueRef};
+use cranelift_entity::EntityRef;
 use log::trace;
-use std::collections::HashMap;
 
-/// Holds the mutable elements of an interpretation. At some point I thought about using
-/// Cell/RefCell to do field-level mutability, thinking that otherwise I would have to
-/// pass around a mutable object (for inst and registers) and an immutable one (for function,
-/// could be self)--in the end I decided to do exactly that but perhaps one day that will become
-/// untenable.
+/// The type used for ensuring [Frame](crate::frame::Frame) entries conform to the expected memory layout.
+pub(crate) type Entries = Vec<Option<DataValue>>;
```

Comment:
Maybe use `SecondaryMap` instead?

Reply:
Although having a newtype for indices would be nice, it isn't clear this is a pure win, since `SecondaryMap` will check if it needs to grow the underlying vec on each insertion since it lazily adds capacity (which we can reserve up front, but I'm not sure LLVM can always optimize those checks away).